src/main/java/me/jellysquid/mods/sodium/mixin/core/MixinMinecraftClient.java

```diff
@@ -17,9 +18,12 @@ public class MixinMinecraftClient {
 
     @Inject(method = "<init>", at = @At("RETURN"))
     private void postInit(RunArgs args, CallbackInfo ci) {
+        var parent = MinecraftClient.getInstance().currentScreen;
+
         if (SodiumClientMod.options().isReadOnly()) {
-            var parent = MinecraftClient.getInstance().currentScreen;
             MinecraftClient.getInstance().setScreen(new ConfigCorruptedScreen(() -> parent));
+        } else if (MissingIndiumScreen.shouldShow()) {
+            MinecraftClient.getInstance().setScreen(new MissingIndiumScreen(() -> parent));
```

Comment:
If the config is corrupted AND indium is missing, it is possible to skip the indium screen. I am not sure how important this is, and whether it is worth addressing.